nixos/modules/tasks/trackpoint.nix

```diff
@@ -68,20 +77,20 @@ with lib;
     (mkIf cfg.enable {
       services.udev.extraRules =
       ''
-        ACTION=="add|change", SUBSYSTEM=="input", ATTR{name}=="TPPS/2 IBM TrackPoint", ATTR{device/speed}="${toString cfg.speed}", ATTR{device/sensitivity}="${toString cfg.sensitivity}"
+        ACTION=="add|change", SUBSYSTEM=="input", ATTR{name}=="${cfg.device}", ATTR{device/speed}="${toString cfg.speed}", ATTR{device/sensitivity}="${toString cfg.sensitivity}"
       '';
 
       system.activationScripts.trackpoint =
         ''
-          ${config.systemd.package}/bin/udevadm trigger --attr-match=name="TPPS/2 IBM TrackPoint"
+          ${config.systemd.package}/bin/udevadm trigger --attr-match=name="${cfg.device}"
         '';
     })
 
     (mkIf (cfg.emulateWheel) {
       services.xserver.inputClassSections =
         [''
         Identifier "Trackpoint Wheel Emulation"
-          MatchProduct "${if cfg.fakeButtons then "PS/2 Generic Mouse" else "ETPS/2 Elantech TrackPoint|Elantech PS/2 TrackPoint|TPPS/2 IBM TrackPoint|DualPoint Stick|Synaptics Inc. Composite TouchPad / TrackPoint|ThinkPad USB Keyboard with TrackPoint|USB Trackpoint pointing device|Composite TouchPad / TrackPoint"}"
+          MatchProduct "${if cfg.fakeButtons then "PS/2 Generic Mouse" else "ETPS/2 Elantech TrackPoint|Elantech PS/2 TrackPoint|${cfg.device}|DualPoint Stick|Synaptics Inc. Composite TouchPad / TrackPoint|ThinkPad USB Keyboard with TrackPoint|USB Trackpoint pointing device|Composite TouchPad / TrackPoint"}"
```

Comment:
What's up with this list of device names? And why is only the `TPPS/2 IBM TrackPoint` in the udev rules above?

Reply:
Looking through the commit history, the udev rules existed first and this section was added with the config from [thinkwiki](https://www.thinkwiki.org/wiki/How_to_configure_the_TrackPoint#Scrolling). Afterwards people added their devices to the xorg config.

IMHO the list here is not necessary and just configuring the device would be nicer, but I wouldn't want to break existing configurations.

Reply:
Hmm alright, but then I think it would make more sense to *append* cfg.device to this list instead of replacing the previously fixed entry. Then the old entry will surely always still be there, which makes sense, because it's compatible, even if cfg.device is set differently.

Reply:
I didn't think it would matter, if somebody uses that device they wouldn't change the cfg but I'll change it so that I append it, it can't hurt :)